src/routers/user-router.ts

```diff
@@ -0,0 +1,130 @@
+import { Request, Response, Router } from 'express';
+import { v4 as uuidv4 } from 'uuid';
+
+import asyncHandler from '../handle-midleware/utils';
+import { validateSchema } from '../validation/validation-middleware';
+
+import * as users from '../../data/data.json';
+import { userSchema } from '../validation/schema';
+
+export const router = Router();
+
+export interface User {
+    id: string;
+    login: string;
+    password: string;
+    age: number;
+    isDeleted: boolean;
+}
+
+const getAutoSuggestUsers = (loginSubstring: string, limit: string) =>
+    users.users
+        .filter((user: User) => !user.isDeleted)
+        .sort((user1: User, user2: User) => {
+            if (user1.login < user2.login) {
+                return -1;
+            }
+            if (user1.login > user2.login) {
+                return 1;
+            }
+            return 0;
+        })
+        .filter((user: User) => user.login.includes(loginSubstring))
+        .slice(0, +limit);
```

Comment:
In my opinion better use filter before sorting. In this way `filter().slice().sort()`. We should not need to sort all data, We should sort only limited data.

Reply:
fixed